### ISSUE TITLE
Datapack about 3.3 and 4.0

### ISSUE THREAD

**Issue by Composer-Hu** [blarp5090/BWSB#32]
In fact, I signed up for Cracker over the summer and qualified for the beta. In the meantime, I keep up with each beta update based on your original data pack, what I have done is of course nothing compared to you, and I did it according to my own personal preference and to the best of my personal ability to reproduce MCSM.

Now I want to share it with you, along with the sound effects I made. Feel free to use them, and I hope this stack of packages helps as you develop new versions!

**Comment by Composer-Hu** [comment]
[[1.20.1] - 4.0.zip](https://github.com/WitherBean/BWSB/files/12569873/1.20.1.-.4.0.zip)
[[1.19.4]-3.3.zip](https://github.com/WitherBean/BWSB/files/12569874/1.19.4.-3.3.zip)

**Comment by Composer-Hu** [comment]
[sounds1.zip](https://github.com/WitherBean/BWSB/files/12569879/sounds1.zip)
[sounds2.zip](https://github.com/WitherBean/BWSB/files/12569880/sounds2.zip)
[sounds3.zip](https://github.com/WitherBean/BWSB/files/12569883/sounds3.zip)
[sounds4.zip](https://github.com/WitherBean/BWSB/files/12569884/sounds4.zip)

**Comment by Composer-Hu** [comment]
[sounds5.zip](https://github.com/WitherBean/BWSB/files/12569891/sounds5.zip)

**Comment by Composer-Hu** [comment]
[sounds6.zip](https://github.com/WitherBean/BWSB/files/12569894/sounds6.zip)

**Comment by Composer-Hu** [comment]
[sounds7.zip](https://github.com/WitherBean/BWSB/files/12569902/sounds7.zip)

**Comment by Composer-Hu** [comment]
[sounds8.zip](https://github.com/WitherBean/BWSB/files/12569903/sounds8.zip)

**Comment by Composer-Hu** [comment]
I'm sorry that I need to upload this many times because of the limitation of file upload size. These are the things I've been working on for a while, I hope you found them helpful!

**Comment by WitherBean** [comment]
Thank you for the sounds, I will be using them for Witherstormmod New Textures 3.5.0. 3.5.0 will be for MCSM 3.3. Do you want to be marked as a developer of BWSB?

**Comment by Composer-Hu** [comment]
Thank you for your kindness, but I just want to share my content to help you, and this help is not large, so I don't expect that I can have any contribution markers.

**Comment by WitherBean** [comment]
Do you have a sounds.json file for all of these sounds? If so, will you send it please?

**Comment by Composer-Hu** [comment]
I'm sorry I forgot this, I packaged it with the WS texture I made and sent it to you.
[wither_storm.zip](https://github.com/WitherBean/BWSB/files/12569963/wither_storm.zip)

**Comment by Composer-Hu** [comment]
[hole.zip](https://github.com/WitherBean/BWSB/files/12569973/hole.zip)
Oh, alos I change your datapack's hole model and texture.

**Comment by Composer-Hu** [comment]
![tractor_beam_outline](https://github.com/WitherBean/BWSB/assets/127498588/4d4a50d7-5f80-473e-b4ac-44f1caff9a38)
This is the tractor_beam_outline texture, I think maybe that's all I can give you right now.

**Comment by WitherBean** [comment]
Thank you, and here is something too:
`{
	"tractor_beams":
	{
		"red": 86,
		"green": 46,
		"blue": 165
	},
	"tractor_beams_night":
	{
		"red": 86,
		"green": 46,
		"blue": 165
	},
	"night_shine":
	{
		"red": 59,
		"green": 172,
		"blue": 189,
		"alpha": 140
	},
	"sky_colors": {
		"5": {
			"day":
			{
				"sky_darken_clouds":
				{
					"red": 130,
					"green": 47,
					"blue": 149
				},
				"sky_darken":
				{
					"red": 123,
					"green": 60,
					"blue": 165
				},
				"sky_darken_fog":
				{
					"red": 61,
					"green": 3,
					"blue": 100
				}
			},
			"night": 
			{
				"sky_darken_clouds":
				{
					"red": 130,
					"green": 47,
					"blue": 149
				},
				"sky_darken":
				{
					"red": 123,
					"green": 60,
					"blue": 165
				},
				"sky_darken_fog":
				{
					"red": 61,
					"green": 3,
					"blue": 100
				}
			}
		},
		"6": {
			"day":
			{
				"sky_darken_clouds":
				{
					"red": 130,
					"green": 47,
					"blue": 149
				},
				"sky_darken":
				{
					"red": 123,
					"green": 60,
					"blue": 165
				},
				"sky_darken_fog":
				{
					"red": 61,
					"green": 3,
					"blue": 100
				}
			},
			"night":
			{
				"sky_darken_clouds":
				{
					"red": 130,
					"green": 47,
					"blue": 149
				},
				"sky_darken":
				{
					"red": 123,
					"green": 60,
					"blue": 165
				},
				"sky_darken_fog":
				{
					"red": 61,
					"green": 3,
					"blue": 100
				}
			}
		},
		"7": {
			"day": 
			{
				"sky_darken":
				{
					"red": 123,
					"green": 60,
					"blue": 165
				},
				"sky_darken_clouds":
				{
					"red": 130,
					"green": 47,
					"blue": 149
				},
				"sky_darken_fog":
				{
					"red": 61,
					"green": 3,
					"blue": 100
				}
			},
			"night": 
			{
				"sky_darken":
				{
					"red": 123,
					"green": 60,
					"blue": 165
				},
				"sky_darken_clouds":
				{
					"red": 130,
					"green": 47,
					"blue": 149
				},
				"sky_darken_fog":
				{
					"red": 61,
					"green": 3,
					"blue": 100
				}
			}
		}
	}
}` I have edited the file to make it look perfect in game.

**Comment by Composer-Hu** [comment]
great!

**Comment by Composer-Hu** [comment]
Regarding datapack, I'm not going to deal with the "packet error" thing. But after entering the game in "safe mode", with the command: /datapack enable (datapack's name) ,still available. 
It is worth noting because healing potions can heal issues with command blocks, so cracker made it immune to all potion effects. The datapack I just sent already uses another method to replace the original "floating" potion effect, but feel free to modify it if you have a better idea.

**Comment by WitherBean** [comment]
I will update BWSB later today, then I will make changes to the version of BWSB that you sent and upload it later (I do not have a time planned for this but it will take take long).

**Comment by Composer-Hu** [comment]
No problem! btw, the new version of WS's mod is even less compatible with optifine. Perhaps you can try the following mods instead, they are generally sufficient: 
https://www.curseforge.com/minecraft/mc-mods/rubidium/files/4573226
https://www.curseforge.com/minecraft/mc-mods/oculus/files/4578741
https://www.curseforge.com/minecraft/mc-mods/entity-model-features
https://www.curseforge.com/minecraft/mc-mods/entity-texture-features-fabric

**Comment by Composer-Hu** [comment]
In 1.20.1, all but 0.6.5 versions(Rubidium) crash the game.
https://www.curseforge.com/minecraft/mc-mods/textrues-rubidium-options
https://www.curseforge.com/minecraft/mc-mods/rubidium-extra
Btw, the above two mods are rubidium extension functions.

**Comment by WitherBean** [comment]
BWSB has been updated to 5.4.0.

**Comment by Composer-Hu** [comment]
8 min ago, so quickly lol.

**Comment by WitherBean** [comment]
I am looking in the datapack's code, and you are really talented with this!

**Comment by Composer-Hu** [comment]
Thank you, but it still has too many flaws, and I spent all summer holiday to fix it, I am far less than you. So I instead of keeping it to myself, I would like to share it with you and make it better!

**Comment by Composer-Hu** [comment]
You are a very powerful, very thoughtful, very creative datapacks author :)

**Comment by WitherBean** [comment]
Thank you!

**Comment by Composer-Hu** [comment]
[functions (new added).zip](https://github.com/WitherBean/BWSB/files/12574380/functions.new.added.zip)
I was just looking at how you made an F-Bomb by discarding items, I expanded them to support discarding items (command_block_tools) onto the anvil and crafting, with particles, sound effects. But I only made diamond tools, and I've also tested that it works, sound effects I sent it earlier. It's called "witherstormmod:command_tools_craft." So if you like the idea, you can just use it :)

**Comment by WitherBean** [comment]
The flaming wither skull kill sound effect does not work in game.

**Comment by Composer-Hu** [comment]
It needs to be paired with a packet to work, the functions folder contains several items such as flaming_wither_skull, which are modified from your "firebal" project.

**Comment by WitherBean** [comment]
What do you mean?

**Comment by Composer-Hu** [comment]
[skulls part.zip](https://github.com/WitherBean/BWSB/files/12576859/skulls.part.zip)
This part is related to the items in the packet that support the effect of this sound.

**Comment by Composer-Hu** [comment]
[sounds.json.txt](https://github.com/WitherBean/BWSB/files/12576886/sounds.json.txt)
And to avoid the problem with the sounds.json file, I resend it.

**Comment by WitherBean** [comment]
The sounds do not appear in game with the /playsound command.

**Comment by Composer-Hu** [comment]
This may be the problem with the sounds.json file, I have resend it.

**Comment by WitherBean** [comment]
It is not in the lang.

**Comment by Composer-Hu** [comment]
sounds.json the file path is assets🔜witherstormmod, the sound effects paht is assets🔜witherstormmod🔜sounds

**Comment by WitherBean** [comment]
None of the new sound effects play in game.

**Comment by Composer-Hu** [comment]
Do you add it to your resource pack? Try the command if you've already done it ：/playsound witherstormmod:wither_flaming_skull_kill hostile @a ~ ~ ~ 300

**Comment by Composer-Hu** [comment]
![~8QDICQ}B8WFS6B)F_B{B`J](https://github.com/WitherBean/BWSB/assets/127498588/9659656f-70c0-4935-8bd0-88d396b97bd2)

**Comment by Composer-Hu** [comment]
![)M_YUK7N1}D8IKG05CSAU$3](https://github.com/WitherBean/BWSB/assets/127498588/79c5ff5f-1ec3-4a12-9e7c-0d44f1b425eb)

**Comment by WitherBean** [comment]
<img width="1304" alt="Screen Shot 2023-09-11 at 6 52 02 AM" src="https://github.com/WitherBean/BWSB/assets/114886300/4e638f68-4562-4b1e-8ebf-c71e9e486dbc">

**Comment by WitherBean** [comment]
<img width="862" alt="Screen Shot 2023-09-11 at 6 53 37 AM" src="https://github.com/WitherBean/BWSB/assets/114886300/e2e457c8-7839-470a-8bb9-84cc528158a9">

**Comment by Composer-Hu** [comment]
What about the other sound effects?

**Comment by WitherBean** [comment]
They don't work either.

**Comment by Composer-Hu** [comment]
That's weird. I think I've had that problem before, my fix is to copy a sound code that currently works, just like cracker already has a working sound code then modify the corresponding sound file name.

**Comment by Composer-Hu** [comment]
![PYDS({KE GKO7%HGLP Z0IP](https://github.com/WitherBean/BWSB/assets/127498588/a98ff44f-0c3b-4cbb-baea-fccbea2a5a95)
![N YRZT 3~IB6(G9(IGR3 3](https://github.com/WitherBean/BWSB/assets/127498588/5e4740f0-0b64-4294-9cbb-5b2c877b50c5)

**Comment by WitherBean** [comment]
I am using 1.19.4, 3.3. Maybe it is that?

**Comment by Composer-Hu** [comment]
Is your file path like the one in my picture? And is the souns.json file used as it is? There may be an error if you are copying and merging into your previous project, any error will cause the entire file to be invalidated.

**Comment by WitherBean** [comment]
Yes.

**Comment by WitherBean** [comment]
<img width="1002" alt="Screen Shot 2023-09-11 at 7 06 42 AM" src="https://github.com/WitherBean/BWSB/assets/114886300/77610d46-d3dd-49d4-92d7-80f2a6081d6e">

**Comment by Composer-Hu** [comment]
This is really weird, I'm trying to package a new resource pack now, but you'll need to put all the audio files into the sounds folder yourself later, making sure the audio is all separate and there's no other folder to wrap them in.

**Comment by Composer-Hu** [comment]
[sound_effets.zip](https://github.com/WitherBean/BWSB/files/12577132/sound_effets.zip)

**Comment by Composer-Hu** [comment]
Please try again to put the individual audio in the last sounds folder and the top resource pack.

**Comment by WitherBean** [comment]
It still does nothing.

**Comment by Composer-Hu** [comment]
I just tried it again and it works on my end, why is there such a strange thing.

**Comment by WitherBean** [comment]
Maybe the whole sounds.json file has do be re-made?

**Comment by WitherBean** [comment]
I am working on this too.

**Comment by Composer-Hu** [comment]
The ones I sent to you are the ones I am currently using, and in theory they are exactly the same.

**Comment by Composer-Hu** [comment]
https://www.mediafire.com/file/qvrrnwmra8bbq7k/sounds.zip/file
I tried to package all the audio completely, now you just need to unpack and replace the original.

**Comment by WitherBean** [comment]
It might be the sound file type. How did you turn them into .ogg?

**Comment by Composer-Hu** [comment]
The audio production software I use is GoldWave, after finishing the production, I will choose to export the audio format as.ogg

**Comment by WitherBean** [comment]
I remade the sounds and they still do not play.

**Comment by WitherBean** [comment]
[sounds.json.zip](https://github.com/WitherBean/BWSB/files/12577323/sounds.json.zip)

**Comment by Composer-Hu** [comment]
I don't know why this is, I have tested all the audio completely using the files you sent, and there is no problem with them. I will re-create a complete audio resource pack, upload the link later, you can use it directly as a resource pack to load, and placed at the top.

**Comment by Composer-Hu** [comment]
https://www.mediafire.com/file/1leyb1mkibulfsl/sound_effets.zip/file
This is exactly the same resource pack I just tested that worked, and my version of the game is 1.20.1
This is already a fully usable resource package

**Comment by WitherBean** [comment]
I am using 1.19.4, It might not work but I will try it.

**Comment by Composer-Hu** [comment]
Normally it is not limited by version, because I only changed the version number in the pack.mcmeta file when I migrated the resource pack, and everything else still works.

**Comment by WitherBean** [comment]
It does not work, just the added sounds to the base.

**Comment by Composer-Hu** [comment]
Perhaps you can try version 1.20.1 first, using this resource pack

>

**Comment by WitherBean** [comment]
1.20.1 has no sound for me normaly.

**Comment by Composer-Hu** [comment]
![6C%IX3GO MTZ`T)OCV{SYT1](https://github.com/WitherBean/BWSB/assets/127498588/de6dfd81-9804-40a8-9115-4545350c2cae)
I even tried to uninstall the WS module, but the sound is still unaffected and it is separate from the resource pack

**Comment by Composer-Hu** [comment]
![@3 H6OORW134JOFC~9ZI5VN](https://github.com/WitherBean/BWSB/assets/127498588/c2422453-4f69-4454-ae00-7fa8c1b78190)

**Comment by WitherBean** [comment]
What mods are installed?

**Comment by Composer-Hu** [comment]
![9V1WBW624 B(S%@O~}BINV1](https://github.com/WitherBean/BWSB/assets/127498588/b9b79b0f-0f84-4b04-8f70-353c7a2af589)
![AC$HX6D$QR@HOGYTP7{6C4U](https://github.com/WitherBean/BWSB/assets/127498588/058b60a0-3cc5-4be7-a605-f4506663cd4a)

**Comment by Composer-Hu** [comment]
![Q T$6HP}`A%31(OB$DG3KMG](https://github.com/WitherBean/BWSB/assets/127498588/1b6cfc27-cbfa-4315-b323-d738ab01b4a4)

**Comment by WitherBean** [comment]
In 1.20.1 along with no sound, the custom sounds are still not there.

**Comment by Composer-Hu** [comment]
This is really bizarre, the file I sent is exactly the same as the normal file I tested.

**Comment by Composer-Hu** [comment]
A more crude approach is to simply replace the sounds.json and sounds in CWSM, exit the game and re-enter. I mean directly modify the resource files in the mod.

**Comment by WitherBean** [comment]
I do not know how to do that. I know how to get into the files and read the .class files, but not edit them.

**Comment by Composer-Hu** [comment]
The resource is not part of the closed source part, it is the same path as the resource package.

**Comment by WitherBean** [comment]
When you exported the sound files, did you use .ogg Vorbis or Opus?

**Comment by Composer-Hu** [comment]
![FPK5HHFJG~OL0RB5H2@7QWD](https://github.com/WitherBean/BWSB/assets/127498588/48ded37b-300a-4795-bf05-2bf4be290508)
I'm not sure which one this is.

**Comment by Composer-Hu** [comment]
![$M_Y1R@}A6GT5NZ$BJFDWXB](https://github.com/WitherBean/BWSB/assets/127498588/265e50ff-0654-48eb-b95b-6ef7a418f3f7)

**Comment by WitherBean** [comment]
Use ogg vorbis for Minecraft.

**Comment by Composer-Hu** [comment]
Yes, that's what I use.

**Comment by WitherBean** [comment]
It works playing different sounds at once to create the explosion sound.

**Comment by Composer-Hu** [comment]
Is it working now?

**Comment by WitherBean** [comment]
Not the custom sounds.

**Comment by Composer-Hu** [comment]
https://www.mediafire.com/file/1leyb1mkibulfsl/sound_effets.zip/file

Did you use this resource pack to load and play custom sounds after entering the game?

**Comment by WitherBean** [comment]
I can load it, but the sounds are still not there.

**Comment by Composer-Hu** [comment]
Is it convenient for you to take a screenshot of the resource pack list? Also screenshot command: /playsound witherstormmod: (The content that pops up automatically after)

**Comment by WitherBean** [comment]
<img width="1627" alt="Screen Shot 2023-09-11 at 8 33 20 AM" src="https://github.com/WitherBean/BWSB/assets/114886300/89646bb0-0f2c-442e-92fa-5164dbc529a3">

**Comment by WitherBean** [comment]
<img width="1250" alt="Screen Shot 2023-09-11 at 8 34 04 AM" src="https://github.com/WitherBean/BWSB/assets/114886300/a9a60b89-520c-45e7-a256-3b8353188b28">

**Comment by WitherBean** [comment]
I could change the default explosion sound of Minecraft.

**Comment by Composer-Hu** [comment]
Did you merge my JSON file with your original file? I remember a time when you wrote a part of the audio wrong and it all didn't work. Unfortunately, I can't remember which part.

**Comment by Composer-Hu** [comment]
[random.zip](https://github.com/WitherBean/BWSB/files/12578015/random.zip)
I've changed it, but it's not working out so well.

**Comment by Composer-Hu** [comment]
https://www.mediafire.com/file/daqc5qmw8raplec/minecraft_-_sounds.zip/file
I've also changed the Minecraft audio quite a bit, which is for the minecraft section.

**Comment by WitherBean** [comment]
Overwriting the Minecraft files works.

**Comment by Composer-Hu** [comment]
It seems that ,I think the problem is with JSON, the audio files are not abnormal.

**Comment by WitherBean** [comment]
Do you have a 1.19.4 sounds.json file?

**Comment by WitherBean** [comment]
Should I do a custom FactionCraft raid for BWSB?

**Comment by Composer-Hu** [comment]
I don't quite understand. And I used the same sounds.json file for both 1.19.4 and 1.20.1.

**Comment by Composer-Hu** [comment]
https://www.mediafire.com/file/m5cit5kxdbvoook/sounds_%2526_tick.zip/file

These are the things I made today that I tested on my side and they all worked.

**Comment by Composer-Hu** [comment]
[supplementary content.zip](https://github.com/WitherBean/BWSB/files/12607220/supplementary.content.zip)
Sorry, I forgot to put their subfiles in the folder last time.

**Comment by Composer-Hu** [comment]
And #cluster sounds, I separated the sound of normal shaking from the sound of the cube breaking.

**Comment by WitherBean** [comment]
Please do not take other people's sounds from their projects.

**Comment by Composer-Hu** [comment]
Sorry but which sounds are you referring to?

**Comment by WitherBean** [comment]
<img width="1004" alt="Screen Shot 2023-09-27 at 6 37 56 AM" src="https://github.com/WitherBean/BWSB/assets/114886300/8c99c71c-253e-4a11-a6a5-9588f18c1780">

**Comment by Composer-Hu** [comment]
I'm so sorry, I will now identify which sound effects are and delete them.

**Comment by Composer-Hu** [comment]
I am very sorry that I did not clean up my sounds folder and caused you trouble, I originally wanted to erase all the sound effects that I didn't make and keep the sound effects that you had/that I later made and added myself. I'd like to offer you a new explosion as a token of my apology, I can totally guarantee that I made them myself, It may be trivial, but I hope to prove my attitude. Sorry again!
[explode.zip](https://github.com/WitherBean/BWSB/files/12739550/explode.zip)
